handlers/retest/ci.go

```diff
@@ -0,0 +1,215 @@
+package retest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	. "github.com/Huawei-PaaS/ci-bot/handlers/types"
+
+	"github.com/golang/glog"
+)
+
+const (
+	TravisCIEndPoint = "https://api.travis-ci.org" //travis-ci endpoint
+	ContentTypeJSON  = "application/json"          //content-type
+	TravisAPIVersion = "Travis-API-Version"        //API version, a Mandatory header field for Travis-CI V3 APIs call
+)
+
+type TravisJobType string
+
+//Travis Build Job Names
+const (
+	Verify      TravisJobType = "verify"   //Travis Build Job Name
+	Crossbuild  TravisJobType = "crossbuild" //Travis Build Job Name
+	Build       TravisJobType = "build"   //Travis Build Job Name
+	Unittest    TravisJobType = "unittest" //Travis Build Job Name
+	Integration TravisJobType = "integration" //Travis Build Job Name
+)
+//Travis Job Id's
+const(
+	JobBuild        = iota + 1
+	JobVerify
+	JobUnittest
+	JobIntegration
+	JobCrossbuild
+)
+
+//Function to handle Get Jobs from TravisCI
+func GetJobIdsFromTravisBuild(BuildRefId string, Token string) []byte {
+	var Reqbody io.Reader
+
+	url := fmt.Sprintf("%s%s/jobs", TravisCIEndPoint, BuildRefId)
+	req, err := http.NewRequest(http.MethodGet, url, Reqbody)
+	if err != nil {
+		glog.Errorf("%v", err)
+	}
+	_, Body := SendHttpReqToCI(req, Token)
+	return Body
+}
+
+//Function to handle Triggering Jobs build in TravisCI
+func TriggerJob(JobBuildId string, Token string) (error, int) {
+	var Reqbody io.Reader
+	var statusCode int
+	url := fmt.Sprintf("%s%s/restart", TravisCIEndPoint, JobBuildId)
+	req, err := http.NewRequest(http.MethodPost, url, Reqbody)
+	if err != nil {
+		glog.Errorf("HTTP request failed: %v", err)
+		return err, statusCode
+	}
+	statusCode, _ = SendHttpReqToCI(req, Token)
+	if statusCode == http.StatusAccepted {
+		glog.Info("Restart Job Build is successfully triggered !!")
+	} else {
+		glog.Info("Restart Job Build is failed to trigger, HttpStatus Code:" , http.StatusAccepted)
+		return nil, statusCode
```

Comment:
return err

Reply:
@sids-b , this cant be a err as sending HTTP request is success, request processing result will  be returned in the statuscode.